diff --git a/node_modules/next/dist/esm/server/next-server.js b/node_modules/next/dist/esm/server/next-server.js
index 58856a7..d871131 100644
--- a/node_modules/next/dist/esm/server/next-server.js
+++ b/node_modules/next/dist/esm/server/next-server.js
@@ -1172,7 +1172,7 @@ export default class NextNodeServer extends BaseServer {
                 });
             } finally{
                 if (hasRequestBody) {
-                    requestData.body.finalize();
+                    await requestData.body.finalize();
                 }
             }
         } else {
diff --git a/node_modules/next/dist/server/next-server.js b/node_modules/next/dist/server/next-server.js
index c5c38e1..547c03f 100644
--- a/node_modules/next/dist/server/next-server.js
+++ b/node_modules/next/dist/server/next-server.js
@@ -1241,7 +1241,7 @@ class NextNodeServer extends _baseserver.default {
                 });
             } finally{
                 if (hasRequestBody) {
-                    requestData.body.finalize();
+                    await requestData.body.finalize();
                 }
             }
         } else {
